bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BType.java

```diff
@@ -133,11 +133,16 @@ public int hashCode() {
     }
 
     public String getName() {
-        return typeName;
+        return typeName == null ? "" : typeName;
     }
 
     public String getQualifiedName() {
-        return pkg == null ? typeName : pkg.toString() + ":" + typeName;
```

Comment:
In cases where the typeName is not present, this would return an invalid value, thus the change.

Reply:
Shall we make getQualifiedName also final?